python/setup.cfg

```
@@ -45,6 +45,8 @@ python_requires = >=3.7
 install_requires =
     mmh3
     singledispatch
+    pyyaml
```

Does this make pyyaml a dependency of the library itself? If so, is there anyway to make it just a “test” dependency or something or will we need it at runtime?

---

We'll need it at runtime. Specifically in the `TableMetadata` class which hasn't been merged in yet but the idea is that reading in a table metadata file will be validated against the open api spec.

---

Are we gonna do that for every file on every read? Or just as a test?

---

It's a great question and I think we can decide on that later. The TableMetadata PR is still a WIP but that class includes this [validate](https://github.com/apache/iceberg/pull/3677/files#diff-0a2dcd19c1079e50703f2010f74a47309468503f3ddd753db3aca9951364810fR236) method. So we may have certain paths where we don't validate (loading a metadata file from a path we got from a catalog) and other paths where we do validate (like before writing+committing a new metadata file), just as an example.

---

I think it would be good to pin it on a version. Currently it will just install the latest greatest, but there might be breaking changes (happens all the time in Python land :)
```suggestion
    pyyaml==6.0
```